packages/sdk/src/oracle/store/has.ts

```diff
@@ -0,0 +1,22 @@
+import Read from "./read";
+// we only ignore this specific error type explicitly emitted from reader class, meaning value is undefined
+import { ignoreExistenceError as ignoreError } from "../errors";
+import type { State } from "../types/state";
+
+// This class checks for existence for values you could potentially read. It mirrors the read interface but returns bools from functions.
+export default class Has {
+  private read: Read;
+  constructor(private state: State) {
+    // by design, reads do not mutate or cause side effects
+    this.read = new Read(state);
+  }
+  inputRequest = (): boolean => {
+    // the rule about reads is they will only throw errors if it cant return the value you request.
+    // If no errors are thrown the read was successful.
+    // Reads will never not return a value without throwing an error, enforced by TS.
+    return !!ignoreError(this.read.inputRequest);
+  };
+  sortedRequestsService = (): boolean => {
+    return !!ignoreError(this.read.sortedRequestsService);
```

Comment:
nit: something to watch out for in this is that you're not just checking for undefined, you're checking for _any_ falsy value, which can capture quite a few reasonable return values for arbitrary functions:
```
> !!"" || !!null || !!undefined || !!0 || !!-0 || !!false
false
```

I might suggest either:
```
ignoreError(...) !== undefined
```

Or changing it to `didThrowExistenceError(...)` and having it explicitly return true or false.


As a secondary note: I think the ignoreExistenceError function name is a little confusing. The fact that it's returning a value is unclear from the name.

Reply:
the existence assertion ` assertExists` checks for null or undefined using ` if (!exists(condition)) `, `exists`does this check under the hood: ` return value !== null && value !== undefined; `.  

> As a secondary note: I think the ignoreExistenceError function name is a little confusing. The fact that it's returning a value is unclear from the name.

hmm yea im not sure what to call it. it converts error to undefined or passes the result of the call back

Reply:
Are 0 values allowed as return values? What about booleans? Empty strings? All of these are falsy, so you may end up with false-positives (meaning you'll think there is no value when there really is one).

Reply:
sorry i see what you are saying now, yes i need to change this, thanks